### PR TITLE
Enable IPv6 rules in the supervisor

### PR DESCRIPTION
**Pull request by 0xmchadha**
This PR modifies the supervisor to allow ipv4 and ipv6 rules to be programmed in the iptables. ipsets, portsets for target networks and external networks are also handled in this PR.

### CODE REVIEW

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "d994744bf00a1eb9ef2e0d0eb8b94eb6f6711184"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "af6819e403c4f4c5d6655064e5221ae17258cfc5"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "42e13c1125d166dfcbf88034359e70b542d44103"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "4a68a283e5b3eae99d1feb6cffef6fccc38a44b3"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "42e13c1125d166dfcbf88034359e70b542d44103"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "4a68a283e5b3eae99d1feb6cffef6fccc38a44b3"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "e03d7c3b04b95bf6d83ef6f1197484255b3fab36"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "db951a7efece17dd70ad1048538cdec036b15e99"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "9b0fd6fa7ea12a005d743482608474e435797d1d"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "d40a4fc9f044caa872974ebad4592a223b80ec5f"
  }
]
```

File: controller/internal/processmon/processmon.go [hunk 1]
```diff
@@ -431,13 +430,13 @@ func contextID2SocketPath(contextID string) string {
...
+func processIOReader(fd io.Reader) {
 	reader := bufio.NewReader(fd)
 	for {
 		str, err := reader.ReadString('\n')
 		if err != nil {
 			return
 		}
-		fmt.Print("[" + contextID + "]:" + str)
+		fmt.Print(str)
 	}
```
Comment (dstiliadis) [inline controller/internal/processmon/processmon.go]: Why are these changes in this PR?

File: controller/internal/supervisor/iptablesctrl/iptables.go [hunk 2]
```diff
@@ -817,3 +857,102 @@ func flushUDPConntrack(networks []string) {
...
+func (i *Instance) CleanUp() error {
+
+	if err := i.iptv4.CleanUp(); err != nil {
+		return err
+	}
+
+	if err := i.iptv6.CleanUp(); err != nil {
+		return err
+	}
+
```
Comment (dstiliadis) [inline controller/internal/supervisor/iptablesctrl/iptables.go]: Cleanups should move on to ipv6 even if there is an error in ipv4. We need to try to clean up as much as we can.

File: controller/internal/supervisor/iptablesctrl/iptables.go [hunk 3]
```diff
@@ -817,3 +857,102 @@ func flushUDPConntrack(networks []string) {
 	// 	}
 	// }
 }
+
+// SetTargetNetworks updates ths target networks. There are three different
```
Comment (dstiliadis) [inline controller/internal/supervisor/iptablesctrl/iptables.go]: Let's move all the instance related structures and methods in a different file. It is very confusing to have them in the same file. The corresponding unit tests must be also in a different file.

Comment (0xmchadha) [reply controller/internal/supervisor/iptablesctrl/iptables.go]: Yes, that will make it more readable.

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "36ad3b3dff3d1efa6bea77626af0e793693e3c1e"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "61005c41e0d1908f6f772bc7126af4d37a3c532b"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "36ad3b3dff3d1efa6bea77626af0e793693e3c1e"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "61005c41e0d1908f6f772bc7126af4d37a3c532b"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "1dbce2b170dfeaa3de472127c4cd0874e9f4ba7a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "863b7a0850444b7cc07311202ad7c54101a15117"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "2f7f56ff5f66e59b2284e455c9a23155456efafa"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "4f350e8da13b634ba145ca0cbd9eb52b900b40df"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "2f7f56ff5f66e59b2284e455c9a23155456efafa"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "2f7f56ff5f66e59b2284e455c9a23155456efafa"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "51d53a362723eb432edae362f3906de7c4cdb704"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "51d53a362723eb432edae362f3906de7c4cdb704"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "51d53a362723eb432edae362f3906de7c4cdb704"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "04405d72b65eab5d0187cb8058f37211e6a0e1f7"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "04405d72b65eab5d0187cb8058f37211e6a0e1f7"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "77bff93acfda651dc6031fb4231e11ae98cc5938"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "efd3ec74e7323dc46bd910de0362b0cd04fca9c4"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "ad944830aab385063c7cdaeff03927bd482ac4d2"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "8f380c3844445b49547966cc91e45670ef059f36"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "66e83e0e28bcab989d7558059aa89623d141eed5"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "66e83e0e28bcab989d7558059aa89623d141eed5"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  },
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "66e83e0e28bcab989d7558059aa89623d141eed5"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "66e83e0e28bcab989d7558059aa89623d141eed5"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "59b6c8c50cc3fd8bd89e468a98eb425463b5c598"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "66e83e0e28bcab989d7558059aa89623d141eed5"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "c2c98634c873fbcf3ce150e00b7e8cbc8f6aab0a"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "c2c98634c873fbcf3ce150e00b7e8cbc8f6aab0a"
  },
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "fbe2ee2e2fd9bb211d8aeb5c226358880555ef32"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "4d0a307250084a9796b61347b9b7034a75f718c0"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "c2c98634c873fbcf3ce150e00b7e8cbc8f6aab0a"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "ba467bce7be951d1993055995d2579ca3360b936"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "c2c98634c873fbcf3ce150e00b7e8cbc8f6aab0a"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "ba467bce7be951d1993055995d2579ca3360b936"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "c2c98634c873fbcf3ce150e00b7e8cbc8f6aab0a"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "ea2cce3f56eb7899ed6fb53c2dbcd21d7f2214ee"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "bdba5fbde8171a9a66d9aaa3f48c6aea2cfff185"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "65df7124c3242b536223062c7c740808ce9cbc39"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "bdba5fbde8171a9a66d9aaa3f48c6aea2cfff185"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "bdba5fbde8171a9a66d9aaa3f48c6aea2cfff185"
  },
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "aeb6b1cf3ed18f07fb869b0aa2a3a5c7189ed073"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "aeb6b1cf3ed18f07fb869b0aa2a3a5c7189ed073"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "7d8514ed17cb4b8fbd8c1db7ad4b4c8dd78f022f"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "d9deafdf1e1d498913b45e1d69dc9c7f85fe089c"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "d9deafdf1e1d498913b45e1d69dc9c7f85fe089c"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "acc484753c5547624659ae0bf343096f4ba7ebab"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "756ae25801ceaa5ff39424f66b6d555a8a7807c7"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "97250ed66570a7805f6f55f98e8a0070ab6486dd"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "5c3d93aeb1ed27eb7ce24933c4ac843e224c3fbc"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "c5a22567e397761ee409e454147d07c7d19323f9"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "647f257806e4459a15a43d75d0148df17dfb3625"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1128",
    "commit-sha": "4d53cbff6da97c6ae24b1ceffed9399e023c060e"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "777",
    "commit-sha": "2e3d988ab9a359c46d0376bac4ca18d7c0203083"
  }
]
```

**Review by dstiliadis — APPROVED** [review]

**Pull request merged**